### PR TITLE
Update composer dependency to fix Redis Key Expiery

### PR DESCRIPTION
**Pull request by toxix**
### Description (*)
I experienced a situation on production system with 15 stores and
organic trafic, where the redis cache grows to over 30GB. It looks like
magento is putting a lot of entries into the redis cache without any
expiary. I needed to limit redis memory to prevent the server from
dying, not how it should be. This had some performance influences.

For this issue there is actualy a fix in place which just need to be
pulled in. This commit is do this. With the fix in place the redis
stabalized at arround 3GB. Over all performance is was also increased
with this change.

Referces:
[1] https://github.com/colinmollenhour/Cm_Cache_Backend_Redis/commit/bc63e7226a525fe95865c756aad0037f0d57d03c
[2] https://github.com/magento/magento2/commit/32058c748731348e7d38cffa7a6c56113e17b3d8
[3] https://github.com/magento/magento2/issues/25487

### Fixed Issues (if relevant)

1. magento/magento2#25487: Redis cache grows unilimmited

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by ihor-sviziev — APPROVED** [review]

**Comment by toxix** [comment]
Hey @buskamuza ,

Can you let me know when do you plan to do the code review? So I can resolve the conflicting composer.lock at that time.

happy coding
Toxix

**Comment by ihor-sviziev** [comment]
Hi @toxix,
It's not mandatory to have feedback from @buskamuza.
Also conflict in composer.lock could be fixed just during merging this PR by internal team.
So no action items for now from your side.

**Comment by engcom-Alfa** [comment]
Hi, @toxix Could you resolve merge conflicts?
 Thanks!

**Review by lbajsarowicz — APPROVED** [review]
https://github.com/colinmollenhour/Cm_Cache_Backend_Redis/compare/1.10.0..1.11.0
No backwards-incompatible changes between the versions.

**Comment by ihor-sviziev** [comment]
Hi @engcom-Alfa,
There is conflict only in "content-hash" in composer.lock. It will appear again and again in case if someone in another PR will change this file.

Could you before testing merge it with accepting any part of "content-hash", and then run `composer update --lock`?

I think it's not worth asking author again and again

**Comment by ihor-sviziev** [comment]
@engcom-Alfa looks like you didn't ran `composer update --lock` as it should update the content-hash in composer.lock, or you didn't pushed your changes

**Comment by slavvka** [comment]
@toxix pleas resolve the conflict

**Comment by slavvka** [comment]
The PR is in the delivery queue. Please be patient.

**Comment by toxix** [comment]
Hey @slavvka ,
 
see comment by @ihor-sviziev  

> There is conflict only in "content-hash" in composer.lock. It will appear again and again in case if someone in another PR will change this file.
> 
> Could you before testing merge it with accepting any part of "content-hash", and then run `composer update --lock`?
> 
> I think it's not worth asking author again and again

So could you please get this PR forward it's sitting around for more than a month already. I don't see why you guys cannot merge this patch or fix the issue yourself. Instead this code is sitting around and get automated comments, while things getting old. Therefore it get's merge conflicts and nothing happens. Pretty annoying.

**Comment by slavvka** [comment]
@toxix It is being delivered in its order. Please be patient.

**Comment by VladimirZaets** [comment]
Please resolve conflicts and check that all tests are passing

**Comment by ihor-sviziev** [comment]
> Please resolve conflicts and check that all tests are passing

@VladimirZaets please see https://github.com/magento/magento2/pull/25488#issuecomment-555911512
This issue will appear again and again until this PR will be merged

**Comment by slavvka** [comment]
@toxix @ihor-sviziev I agree that this file should be regenerated before delivery but we added additional tests and to make sure they are passing we need to have this conflict resolved once more time.

**Comment by engcom-Foxtrot** [comment]
@slavvka updated.

**Review by lbajsarowicz — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-6247](https://jira.corp.magento.com/browse/ENGCOM-6247) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@toxix thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

File: composer.json
Comment (ihor-sviziev) [inline composer.json]: @engcom-Foxtrot why did you replaced dependency for zend-http? I think it's not correct

Comment (engcom-Foxtrot) [reply composer.json]: Hello, @ihor-sviziev. Seems like 2.10.1 makes some backward incompatible changes.

**Comment by slavvka** [comment]
@engcom-Foxtrot I think we need to regenerate `composer.lock` only for changed package with something like `composer update colinmollenhour/cache-backend-redis` in order to avoid bumping versions in other packages

**Pull request merged**
